lib/fog/libvirt/compute.rb

```diff
@@ -120,16 +120,16 @@ def enhance_uri(uri)
           # on macosx, chances are we are using libvirt through homebrew
           # the client will default to a socket location based on it's own location (/opt)
           # we conveniently point it to /var/run/libvirt/libvirt-sock
-          # if no socket option has been specified explicitly
+          # if no socket option has been specified explicitly and
+          # if the socket exists
 
-          if RUBY_PLATFORM =~ /darwin/
+          socketpath="/var/run/libvirt/libvirt-sock"
```

Comment:
This could also be a constant but not big deal.

Reply:
I don't have a strong opinion on this and my Ruby is quite rusty, but if I was to extract this into a constant then we'd probably also need to come up with a different name that adds more context in the global space - for any avoidance of doubt on what it is.

I also think that this particular piece of code looks like small feature and as such it may preferable to keep all the logic close, else a decoupled constant may be forgotten about during future refactoring or removal.